security/pkg/pki/signingapi/metadata.go

```diff
@@ -0,0 +1,68 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signingapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/types"
+)
+
+// Metadata the certificate singing metadata carry from xds node
+type Metadata struct {
+	WorkloadName string   `json:"WorkloadName"`
+	ClusterID    string   `json:"ClusterID"`
+	WorkloadIPs  []string `json:"WorkloadIPs"`
+}
+
+// ParseMetadataProtoStruct return Metadata struct from proto message
```

Comment:
I don't see where this is used? unless github is hiding something...

Reply:
We integrate Keyfactor PKI into Istio, our customers need WorkloadIPs, WorkloadName, etc. to track and monitor

Reply:
So you are importing istio/istio and depending on this function?

Reply:
same question here. we should not be parsing the metadata from the proxy. Just take that struct field and shove it into the outbound call. Why bother parsing the struct and re-encoding? this metadata will keep expanding (some vendors may want much more than this, while others may jsut care about one or two). So we need to keep this as opaque as possible to istiod.

Reply:
yeah the more i read it, it feels like utility code used by the vendors. This doesn't have anything to do with istio.